src/sound.c

```diff
@@ -4,20 +4,26 @@
 #define AMPLITUDE 127
 #define FREQUENCY 44100
 
-void audio_callback(void *user_data, Uint8 *stream, int len) {
-    SDL_memset(stream, 0, len);
-    (void) user_data; // tell the compile that user_data is not used
+static void audio_callback(void *user_data, Uint8 *stream, int len) {
+    if (len < 0) {
+        fprintf(stderr, "len cannot be negative\n");
+    } else if (sizeof(int) > sizeof(size_t) && len > SIZE_MAX) {
+        fprintf(stderr, "len %d is larger than allowed %lu\n", len, SIZE_MAX);
+    }
+
+    SDL_memset(stream, 0, (size_t) len);
+    (void) user_data; // tell the compiler that user_data is not used
 
     for (int i = 0; i < len; i++) {
-        stream[i] = (AMPLITUDE * sin(i * PI * 2 * 604.1 / FREQUENCY));
+        stream[i] = (Uint8) (AMPLITUDE * sin(i * PI * 2 * 604.1 / FREQUENCY));
```

Comment:
Could you please use `_t` suffix types like so
```suggestion
        stream[i] = (uint8_t) (AMPLITUDE * sin(i * PI * 2 * 604.1 / FREQUENCY));
```
as this achieves consistency.
Additionally, Uint* are not C-standard, while uint*_t are :)

Reply:
Sure, I can definitely do that.
Uint8 is actually standard, since it is just a typedef: `typedef uint8_t Uint8;` (see SDL_stdinc.h line 196).

The SDL wiki proposes these types for the callback (see [here](https://wiki.libsdl.org/SDL_AudioSpec)):
```
void SDL_AudioCallback(void*  userdata,
                       Uint8* stream,
                       int    len)
```

In my personal opinion I agree with you; using uint8_t in this case does no harm and achieves consistency